pkg/mappings/resource-mappings.go

```diff
@@ -0,0 +1,31 @@
+package mappings
+
+import (
+	"context"
+
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ResourceMappingsFinder provides functionality of retrieving Resource Mapping CRs
+type ResourceMappingsFinder struct {
+	client client.Client
+}
+
+// NewResourceMappingsFinder creates new ResourceMappingsFinder configured with given client
+func NewResourceMappingsFinder(client client.Client) ResourceMappingsFinder {
+	return ResourceMappingsFinder{
+		client: client,
+	}
+}
+
+// GetResourceMapping retrieves current version of a resource mapping CR with given namespaced name
+func (m *ResourceMappingsFinder) GetResourceMapping(namespacedName types.NamespacedName) (*v2vv1alpha1.ResourceMapping, error) {
+	instance := v2vv1alpha1.ResourceMapping{}
```

Comment:
why not to create a pointer here:

```
instance := &v2vv1alpha1.ResourceMapping{}
```

We would not need to make it a pointer in line 26 and 30.

Reply:
But having it as it is, we explicitly show that we're passing a reference, not value in line 26 and up the call stack in line 30. It seems to be an advised pattern in Go.

Reply:
It would be great if you could point me to this pattern well defined.

I see that value vs pointer semantics for return value seems to be inconsistent between `NewResourceMappingsFinder` and `GetResourceMapping`.

Reply:
Sorry for paid content, but here it is: https://learning.oreilly.com/videos/ultimate-go-programming/9780135261651/9780135261651-UGP2_01_02_03_03. In the transcript starting from "So, there's a guideline here that I want to provide because the ampersand is a very powerful readability operator, and you don't wanna walk away from it"

I agree, there is a return inconsistency between here and the `NewResourceMappingsFinder` and having in mind the advice above we should probably always return a pointer (and show it explicitly on the return line). 

All in all, I think that's something we should discuss and agree on and follow one convention - whichever it is. What is your opinion on returning pointer vs. value?

Reply:
I think it would be best to find go guidelines and follow them. Go was always strong about best practices.

Reply:
What I've found so far is not really a rule, but a hint what might be proper:
- return value;
- return pointer if the returned struct is "big" (when the heap allocation cost gets lower than copying value);
Source: https://philpearl.github.io/post/bad_go_pointer_returns/

Reply:
Yes, it makes sense.